jgrapht-core/src/main/java/org/jgrapht/graph/Subgraph.java

```diff
@@ -459,6 +419,49 @@ public void setEdgeWeight(E e, double weight)
         ((WeightedGraph<V, E>) base).setEdgeWeight(e, weight);
     }
 
+    private void initialize(Set<? extends V> vertexFilter, Set<? extends E> edgeFilter)
+    {
+        // add vertices
+        if (vertexFilter == null) {
+            base.vertexSet().stream().forEach(v -> vertexSet.add(v));
+        } else {
+            if (vertexFilter.size() > base.vertexSet().size()) {
+                base.vertexSet().stream().filter(v -> vertexFilter.contains(v)).forEach(
+                    v -> vertexSet.add(v));
+            } else {
+                vertexFilter.stream().filter(v -> v != null && base.containsVertex(v)).forEach(
+                    v -> vertexSet.add(v));
+            }
+        }
+
+        // add edges
+        if (edgeFilter == null) {
```

Comment:
Can you add one more case:
if( vertexFilter == null && edgeFilter==null) //Subgraph is used as a copy
  edgeSet.addAll(base.edgeSet);